sdk/python/feast/infra/offline_stores/file.py

```diff
@@ -210,6 +213,89 @@ def evaluate_historical_retrieval():
         job = FileRetrievalJob(evaluation_function=evaluate_historical_retrieval)
         return job
 
+    @staticmethod
+    def get_historical_features_by_view(
+        config: RepoConfig,
+        feature_views: List[FeatureView],
+        feature_refs: List[str],
+        entity_df: Union[pd.DataFrame, str],
+        registry: Registry,
+        project: str,
+        start_date: Optional[datetime],
+        end_date: Optional[datetime],
+        full_feature_names: bool,
+    ):
+        if not (isinstance(entity_df, pd.DataFrame) or isinstance(entity_df, str)):
+            raise ValueError(
+                f"Please provide an entity_df of type {type(pd.DataFrame)} instead of type {type(entity_df)}"
+                f"Please provide an entity_df of type {type(pd.DataFrame)} or {type(str)} instead of type {type(entity_df)}"
+            )
+
+        if isinstance(entity_df, str):
```

Comment:
Are you assuming that a `str` is a reference to a feature view? What if the user provides a SQL query?

Reply:
Is a SQL query relevant for file-based offline stores?

Reply:
> Is a SQL query relevant for file-based offline stores?

It isn't, but what happens when we generalize this code to other feature stores? Are you suggesting we'd have different APIs for different offline stores?

Reply:
It could be a query. For the other offline stores the parameter could be "entity_df_or_query"? I'm rather indifferent towards this. 

Happy to go with any approach you think is sensible.

Reply:
I think what is confusing to me is why `get_historical_features_by_view` even supports Pandas/Query if it's by view. What if we just supported a string reference (or even an object reference) to a feature view, and nothing else?

Reply:
Yes, let's do that. It removes the ambiguity.